src/GraphWebhooks/Controllers/ListenController.cs

```diff
@@ -48,8 +43,12 @@ public class ListenController : Controller
         _logger = logger ?? throw new ArgumentException(nameof(logger));
         _ = configuration ?? throw new ArgumentException(nameof(configuration));
 
-        _appIds = new List<Guid> { new Guid(configuration.GetValue<string>("AzureAd:ClientId")) };
-        _tenantIds = new List<Guid> { new Guid(configuration.GetValue<string>("AzureAd:TenantId")) };
+        var appId = configuration.GetValue<string>("AzureAd:ClientId") ??
```

Comment:
```suggestion
        var appId = configuration.GetValue<string>("AzureAd:ClientId") is string appIdValue && !string.IsNullOrEmpty(appIdValue) ? appIdValue :
```

(null coalesce of strings can lead to unexpected behaviours with NRT)
(repeat for other settings)